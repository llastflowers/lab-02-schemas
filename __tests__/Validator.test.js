const { Validator } = require('../lib/Validator.js');

describe('Validator', () => {
    let validator;

    describe('required fields', () => {
        beforeAll(() => {
            validator = new Validator('name', {
                type: String,
                required: true
            });
        });

        it('returns the field of the object', () => {
            const cat = {
                name: 'willow',
                age: 3,
                color: 'black and white'
            };
    
            expect(validator.validate(cat)).toEqual('willow');
        });
    });
});
