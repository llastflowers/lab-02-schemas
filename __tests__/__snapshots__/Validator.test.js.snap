// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator required fields throws an error if a required field is missing 1`] = `"Required field name not found"`;
